doc/releases/release-notes-4.2.rst

```diff
@@ -771,3 +771,9 @@ Other notable changes
 
 * Updated TF-M to version 2.1.2 (from 2.1.1). The release notes can be found at:
   https://trustedfirmware-m.readthedocs.io/en/tf-mv2.1.2/releases/2.1.2.html
+
+* As of commit f726cb5123eda12feb8746e3acb34a1f7d7b0340, Zephyr now uses the ``CMSIS_6`` module
+  (instead of ``cmsis`` which was CMSIS 5.9.0) for Cortex-M architectures.
+  Other modules still relying on the older ``cmsis`` module are encouraged to migrate to ``CMSIS_6``.
+  The path to the ``CMSIS_6`` module can be accessed via the ``CONFIG_ZEPHYR_CMSIS_6_MODULE_DIR`` configuration.
+  Note: Zephyr will continue using the older ``cmsis`` module for Cortex-A and Cortex-R targets.
```

Comment:
I think this should be moved to the migration guide. Release notes I say it should be more of a "new shiny features" document, I would not bother mentioning it here just on the migration guide. No need to mention the commit either, the document is the whole release. Wording sounds a bit funny, you are not exactly encouraged, it's more like you must migrate.

Keep it simple, straight to the point, something like: Cortex-M SoCs now require the cmsis_6 module to build properly.